pkg/template/template.go

```diff
@@ -254,7 +223,51 @@ func (t *dirTemplate) Execute(dirPrefix string) error {
 		}
 
 		return nil
-	})
+	}); err != nil {
+		return err
+	}
+
+	return t.saveContext(localContext)
+}
+
+// updateContext will augment the current context with the values from the file at the provided path
+func (t *dirTemplate) updateContext(path string) error {
+	context, err := readContext(path)
+	if err != nil {
+		return err
+	}
+
+	if context != nil {
+		for key, value := range context {
+			if key == boilr.DelimsProperty {
+				continue
+			}
+
+			// If the value exists in the local file but not in the actual context, we'll just ignore it as most likely the template has changed since it was last saved
+			if _, ok := t.Context[key]; ok {
```

Comment:
I believe `t.Context` can be nil (file doesn't exist returns nil, nil).  Unless it is tangentially checked due to the path context not being nil?

Reply:
Nevermind, it seems that go handles nil maps specially, I did not realize.  Not sure how you can return a nil map but still use it...seems magicy to me :D

Reply:
You can perform read operations on a nil map but cannot set values. You can also perform read operations on a nil slice (although indexing it will get an index out of range because that's just one of the features of a slice, `nil` or not) but you can `len(nilSlice)` or `append(nilSlice, "something")` and get back a single element slice.